include/drivers/emul.h

```diff
@@ -98,6 +98,16 @@ extern const struct emul __emul_list_end[];
 int emul_init_for_bus_from_list(const struct device *dev,
 				const struct emul_list_for_bus *list);
 
+/**
+ * @brief Find an emulator by name
+ *
+ * This function behaves similar to device_get_binding, but for emulators.
+ *
```

Comment:
Could you note where the 'name' comes from? That would be helpful for people I think.

Also for the commit subject, how about 'emul: Support getting an emulator by name?' as emul_get_binding() seems a bit unclear/specific

Reply:
Done, I paraphrased the docs from `device_get_binding()` to keep things as consistent as possible.